httpclient5/src/main/java/org/apache/hc/client5/http/ssl/DefaultHostnameVerifier.java

```diff
@@ -355,7 +355,26 @@ static List<SubjectName> getSubjectAltNames(final X509Certificate cert, final in
                         if (o instanceof String) {
                             result.add(new SubjectName((String) o, type));
                         } else if (o instanceof byte[]) {
-                            // TODO ASN.1 DER encoded form
+                            final byte[] bytes = (byte[]) o;
+                            final String decodedValue;
+                            if (type == SubjectName.IP) {
+                                if (bytes.length == 4) {
+                                    decodedValue = byteArrayToIp(bytes); // IPv4
+                                } else if (bytes.length == 16) {
+                                    decodedValue = byteArrayToIPv6(bytes); // IPv6
+                                } else {
+                                    throw new IllegalArgumentException("Invalid byte length for IP address: " + bytes.length);
+                                }
+                            } else if (type == SubjectName.DNS) {
+                                throw new IllegalArgumentException("Unexpected byte[] for DNS SAN type");
```

Comment:
@arturobernalg `IllegalArgumentException` looks really wrong here. Can we just ignore those bits that we are not able to properly recognize and handle?